src/models/common/index.ts

```diff
@@ -26,9 +26,11 @@ export interface IssuedCurrencyAmount extends IssuedCurrency {
 export type Amount = IssuedCurrencyAmount | string;
 
 export interface Signer {
-  Account: string;
-  TxnSignature: string;
-  SigningPubKey: string;
+  Signer: {
+    Account: string;
+    TxnSignature: string;
+    SigningPubKey: string;
+  };
```

Comment:
this is weird...why are you nesting it like this?

Reply:
That's how the XRPL does it

Reply:
This mirrors the request format (See the example here: https://xrpl.org/submit_multisigned.html#submit_multisigned) - An alternative is pulling the `Signer:` property out (Ex. `Signers: Array<{ Signers: Signer }>` instead of `Array<Signer>`)

Having it in the SIgner interface makes it clear how to construct a Signer (And makes forming a Transaction's typing for it look cleaner), but it makes it less intuitive to access Signer's fields.

I'm leaning towards this encapsulation (With `Signer:` within the type) because it ensures you don't have something like `const signer: { Signer: Signer } = tx.Signers[0]` which feels awkward.

What are your thoughts? (Am I missing a better way to match the Transaction interface?)

Reply:
I think `Memo`s are/should also be nested like this (?)

Whatever we decide, I suppose the ideal is for `Signer` and `Memo` to sort of be consistent with each other

Reply:
Yeah, memos are also like this. If we add a testcase with memos then it should fail

Reply:
If you don't want to fix this in this PR, add a `TODO` somewhere

Reply:
Added a TODO for Memos